### PR TITLE
New Resource `azurerm_virtual_desktop_application`

### PR DESCRIPTION
**Pull request by alexwilcox9**
Resolves https://github.com/terraform-providers/terraform-provider-azurerm/issues/11722

I'll mark this no longer a draft when I have tests and documentation in place. 
Also do I need the schema migration functions? I added them as all the other desktop virtualization resources have one but I wasn't sure if this is needed for a brand new resource?

Thanks

### CODE REVIEW

**Comment by SebRosander** [comment]
@alexwilcox9 do you know when you will have time to finish this off? Have a use-case for this and I'm eager to get it into the provider and don't want to supersede your work by opening yet another PR for this :)

**Comment by alexwilcox9** [comment]
@SebRosander hopefully won't be too long now, just added some tests and they are passing which is a good start
Just as a heads up I'm not looking to cover off MSIX applications in this PR as that needs an MSIX package resource created as well

```
$ TF_ACC=1 go test -v -timeout=2h ./azurerm/internal/services/desktopvirtualization/virtual_desktop_application_resource_test.go
2021/06/15 17:50:37 [DEBUG] not using binary driver name, it's no longer needed
2021/06/15 17:50:37 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccVirtualDesktopApplication_basic
=== PAUSE TestAccVirtualDesktopApplication_basic
=== RUN   TestAccVirtualDesktopApplication_complete
=== PAUSE TestAccVirtualDesktopApplication_complete
=== RUN   TestAccVirtualDesktopApplication_update
=== PAUSE TestAccVirtualDesktopApplication_update
=== RUN   TestAccVirtualDesktopApplication_requiresImport
=== PAUSE TestAccVirtualDesktopApplication_requiresImport
=== CONT  TestAccVirtualDesktopApplication_basic
=== CONT  TestAccVirtualDesktopApplication_requiresImport
=== CONT  TestAccVirtualDesktopApplication_complete
=== CONT  TestAccVirtualDesktopApplication_update
--- PASS: TestAccVirtualDesktopApplication_complete (105.53s)
--- PASS: TestAccVirtualDesktopApplication_basic (106.64s)
--- PASS: TestAccVirtualDesktopApplication_requiresImport (123.00s)
--- PASS: TestAccVirtualDesktopApplication_update (180.66s)
PASS
ok      command-line-arguments  180.764s
```

Just got to do the documentation and I'll mark this ready for review

**Comment by SebRosander** [comment]
@alexwilcox9 Amazing! Great work, really appreciated. No worries about MSIX, we are not there yet.

File: website/docs/r/virtual_desktop_application.html.markdown [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+* `name` - (Required) The name of the Virtual Desktop Application. Changing the name forces a new resource to be created.
+
+* `application_group_id` - (Required) Resource ID for a Virtual Desktop Application Group to associate with the
+    Virtual Desktop Application. Changing the ID forces a new resource to be created.
+
+* `friendly_name` - (Optional) Option to set a friendly name for the Virtual Desktop Application.
+
+* `description` - (Optional) Option to set a description for the Virtual Desktop Application.
+
+* `file_path` - (Required) The file path location of the app on the Virtual Desktop OS.
```
Comment (katbyte) [inline website/docs/r/virtual_desktop_application.html.markdown:81]: would this make more sense as
```suggestion
* `path` - (Required) The file path location of the app on the Virtual Desktop OS.
```

as file seems reduntant?

File: website/docs/r/virtual_desktop_application.html.markdown [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+* `command_line_setting` - (Required) Specifies whether this published application can be launched with command line arguments provided by the client, command line arguments specified at publish time, or no command line arguments at all. Possible values include: `DoNotAllow`, `Allow`, `Require`.
```
Comment (katbyte) [inline website/docs/r/virtual_desktop_application.html.markdown:83]: i think we can just use this to be more consistant
```suggestion
* `command_line_argument_policy` - (Required) Specifies whether this published application can be launched with command line arguments provided by the client, command line arguments specified at publish time, or no command line arguments at all. Possible values include: `DoNotAllow`, `Allow`, `Require`.
```

so the tf becomes `command_line_argument_policy = allow`

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanskk for the PR @alexwilcox9! overall this looks good, i've left some comments inline and no, we don't need a migration for a new resource :) i think this will be good to merge once my comments are addressed and tests pass!

**Comment by alexwilcox9** [comment]
Thanks for the review @katbyte I've made the changes you suggested and resolved the linting issue
Tests are still looking good, hopefully this is good to go now
```
$ TF_ACC=1 go test -v -timeout=2h ./azurerm/internal/services/desktopvirtualization/virtual_desktop_application_resource_test.go 
2021/06/16 11:44:43 [DEBUG] not using binary driver name, it's no longer needed
2021/06/16 11:44:43 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccVirtualDesktopApplication_basic
=== PAUSE TestAccVirtualDesktopApplication_basic
=== RUN   TestAccVirtualDesktopApplication_complete
=== PAUSE TestAccVirtualDesktopApplication_complete
=== RUN   TestAccVirtualDesktopApplication_update
=== PAUSE TestAccVirtualDesktopApplication_update
=== RUN   TestAccVirtualDesktopApplication_requiresImport
=== PAUSE TestAccVirtualDesktopApplication_requiresImport
=== CONT  TestAccVirtualDesktopApplication_basic
=== CONT  TestAccVirtualDesktopApplication_requiresImport
=== CONT  TestAccVirtualDesktopApplication_update
=== CONT  TestAccVirtualDesktopApplication_complete
--- PASS: TestAccVirtualDesktopApplication_complete (108.81s)
--- PASS: TestAccVirtualDesktopApplication_basic (109.07s)
--- PASS: TestAccVirtualDesktopApplication_requiresImport (128.60s)
--- PASS: TestAccVirtualDesktopApplication_update (187.11s)
PASS
ok      command-line-arguments  187.223s
```

**Review by katbyte — APPROVED** [review]
Thanks @alexwilcox9 - LGTM 🍰

**Pull request merged**